server/plugins/Slack.ts

```diff
@@ -1,36 +1,125 @@
-import { APIReturn } from "./APIReturn";
-import { Plugin, GenericNotifier } from "./GenericNotifier";
+import * as querystring from "querystring";
+import fetch from "node-fetch";
+import {Plugin, GenericNotifier} from "./GenericNotifer";
+import APIReturn from "./APIReturn";
 
 interface Config {
 	channels: string[];
 }
 
 class Slack implements GenericNotifier<Config> {
+    private token: string;
+    private domain: string;
 
-	public async check(config: any): Promise<Config> {
-		const channelsCheck: boolean =
-			Array.isArray(config.channels) &&
-			(config.channels as any[]).every(channel =>
+    constructor() {
+        const token = process.env.SLACK_TOKEN;
+        const domain = process.env.SLACK_DOMAIN;
+
+        if (!token) {
+            console.error("Missing Slack token env var");
+        }
+        if (!domain) {
+            console.error("Missing Slack domain env var");
+        }
+
+        if (!token || !domain) {
+			throw new Error("Slack env vars missing, aborting...");
+		}
+
+        this.token = token;
+        this.domain = domain;
+    }
+
+    public async sendMessage(message : string, config : any) : Promise<APIReturn> {
+        await Promise.all(config.channels.map(chan => (async () => {
+            const msg = {
+                token: this.token,
+                channel: chan,
+                as_user: false,
+                username: "HackGT",
+                text: message
+            };
+
+            const qs = querystring.stringify(msg);
+
+            const res = await fetch(`https://${this.domain}.slack.com/api/chat.postMessage?${qs}`);
+            const json = await res.json();
+            let js = JSON.parse(json);
+            for (var a in js) {
+                if (!js.hasOwnProperty(a)) continue;
+                if (a == 'ok') {
+                    if (js[a] == false) {
+                        return [{
+                            error: true,
+                            key: "Slack Master",
+                            message: js[a + 1]
+                        }];
+                    }
+                }
+            }
+            return [{
+                error: false,
+                key: "Slack Master",
+                message: "Message posted at channel: " + chan
+            }];
+        })));
+    }
+
+    public async setup(): Promise<void> {
+        return;
+    }
+
+    public async check(config : any): Promise<Config> {
+        var params = {
+            token: this.token,
+            exclude_archived: true,
+            exclude_members: true
+        };
+
+        const qs = querystring.stringify(params);
+
+        this.instanceOfConfig(config);
```

Comment:
I would make `instanceOfConfig` return a `Config` type so you can assign it and make use of the types, e.g.:
```typescript
const config = this.instanceOfConfig(rawConfig);
```